src/dask_awkward/lib/core.py

```diff
@@ -1530,22 +1564,43 @@ def map_partitions(
     This is effectively the same as `d = c * a`
 
     """
-    token = token or tokenize(fn, *args, meta, **kwargs)
-    label = label or funcname(fn)
+    token = token or tokenize(base_fn, *args, meta, **kwargs)
+    label = label or funcname(base_fn)
     name = f"{label}-{token}"
+    kwarg_deps, kwarg_repacker = unpack_collections(kwargs, traverse=traverse)
+    deps, _ = unpack_collections(*args, *kwargs.values(), traverse=traverse)
+
+    arg_deps_expanded = []
+    arg_repackers = []
+    arg_lens_for_repackers = []
+    for arg in args:
+        this_arg_deps, repacker = unpack_collections(arg, traverse=traverse)
+        if len(this_arg_deps) > 0:
```

Comment:
just for my own clarification: does this case just mean that the `arg` is not a dask collection, correct?

Reply:
Almost, it means arg does not contain any dask collections (where the collection could be arg itself) so we should not repack it. This is what lets the blockwise Index arguments to continue to work as normal, since instead of packing it up we ship it.